src/rdiff_backup/rorpiter.py

```diff
@@ -163,27 +163,26 @@ def finish_processing(self):
             self.branches[-1].gather_from_child(to_be_finished)
 
     def _finish_branches(self, index):
-        """Run Finish() on all branches index has passed
+        """
+        Run end_process_directory() on all branches index has passed
 
         When we pass out of a branch, delete it and process it with
         the parent.  The innermost branches will be the last in the
-        list.  Return None if we are out of the entire tree, and 1
+        list.  Return True if we are out of the entire tree, and False
         otherwise.
-
         """
-        branches = self.branches
         while 1:
-            to_be_finished = branches[-1]
+            to_be_finished = self.branches[-1]
             base_index = to_be_finished.base_index
             if base_index != index[:len(base_index)]:
                 # out of the tree, finish with to_be_finished
                 to_be_finished.end_process_directory()
-                del branches[-1]
-                if not branches:
-                    return None
-                branches[-1].gather_from_child(to_be_finished)
+                del self.branches[-1]
+                if not self.branches:
+                    return True
+                self.branches[-1].gather_from_child(to_be_finished)
             else:
-                return 1
+                return False
 
```

Comment:
I am not quite sure about the changes in this function `_finish_branches`. Is the semantic changed because of `to_be_finished.end_process_directory()`? We substitute for 1 ~> False and for None ~> True.

Reply:
Yes, it's on purpose, it makes the code `if _finish_branches(etc...)` more logical.

Reply:
Then everything looks fine from my side :)